sig-robotics/proposal/Definition of cloud robot CRD and Operator/Definition of cloud robot CRD and Operator.md

```diff
@@ -0,0 +1,157 @@
+- Extending the robot API to enable robot control
+
+Operator is designed to achieve
+
+- automatizing  robotic operations
+- Monitoring robot status
+- automatizing fault handling
+- adaptive controlling
+
+## Proposal
+
+The CRDs and Controllers is designed to provide more flexible and scalable management solutions for cloud robots connected to the KubeEdge platform.
+
+project scope：
+
+The scope of this project includes
+
+- Providing a series of CRDs  and an Operator for managing cloud robotics based on KubeEdge architecture
+  - CRDs to define robot properties and states
+  - CRDs to extend teh robot API, through which we can control the robot
+  - A series of controllers to realize the automatic control of the robot
+- This project is to design the robot CRD based on the interface definition of industrial mobile robots proposed by the Mobile Robot Industry Alliance
+
+targeting users：
+
+- Developers：Building a complete solution for cloud robot management based on KubeEdge
+- End users：Publishing tasks according to requirements and realizing automatic robot control
+
+## Design Details
+
+### Architecture
+
+For this project, the architecture adopted basically follows the overall architecture of KubeEdge. The structure diagram of the project is shown in the figure below
+
+<img src="./images/architecture.png" alt="architecture" style="zoom:90%;" />
+
+The current predominant approach for managing cloud robots is to treat the robot as a unified entity, considering it as a Device and accessing the KubeEdge platform for management and scheduling purposes.
+
+However, given that a robot is inherently composed of diverse sensors and controllers, it is more appropriate to view it as a combination of heterogeneous components rather than an indivisible arrangement. Consequently, in this solution, a robot is treated as an edge node, with its sensors and controllers serving as the device access platform for that node.
+
+## CRD design
+
+The purpose of this proposal is to propose a set of universal CRD and Operator for cloud robot. Since most industrial robot companies in China have joined the Mobile Robot Industry Alliance, the data interface specifications for robots and their scheduling systems proposed by them are widely applicable. This proposal refers to this specification to design the robot CRD.
+
+To achieve our goal, we need three kinds of CRDs: **robot CRD**, **task CRD** and **robotSync CRD**. The robot CRD and task CRD describe the robot status and task information respectively, and the robotSync CRD is used to record the number of robot CR and their configuration informations. When the key information of the robot changes, the status information must be reported immediately, otherwise the information is reported in the form of a heartbeat frame at a period set by the user. Task resources are created by users and delivered by the Controller.
+
+- The fields contained in the **robot** resource can be described as the table below:
+
```

Comment:
**Suggested Addition**: Enhancing Robot CRD with Sensor-Related Fields

To ensure accurate assignment of sensor-related tasks, it is highly recommended to enrich the Robot CRD with sensor-related fields. These additional fields will enable more precise task allocation, taking into account the capabilities and availability of the robot's sensors. It can be an array including the sensor list that you think is optional for the agv/amr robot.

![robot Architecture](https://raw.githubusercontent.com/robolaunch/trademark/main/repository-media/cloudy/images/Architecture.png)